rotkehlchen/exchanges/binance.py

```diff
@@ -522,30 +526,108 @@ def _deserialize_asset_movement(self, raw_data: Dict[str, Any]) -> Optional[Asse
+            query_schema = 'withdrawList'
+        else:
+            raise AssertionError(f'Unexpected binance method case: {method}.')
+
+        results: List[Dict[str, Any]] = []
+        # Create required time references in milliseconds
+        time_delta_ts_ms = time_delta_ts * 1000 - 1  # less than in ms
+        end_ts_ms = end_ts * 1000
+        if start_ts == Timestamp(0):
+            start_ts_ms = Timestamp(BINANCE_LAUNCH_TS) * 1000
+        else:
+            start_ts_ms = start_ts * 1000
+
+        from_ts_ms = start_ts_ms
+        to_ts_ms = (
+            start_ts_ms + time_delta_ts_ms  # Case request with time delta
+            if end_ts_ms - start_ts_ms > time_delta_ts_ms
+            else end_ts_ms  # Case request without time delta (1 request)
+        )
+        while True:
+            options = {
+                'timestamp': ts_now_in_ms(),
+                'startTime': from_ts_ms,
+                'endTime': to_ts_ms,
+            }
+            result = self.api_query_dict(method, options=options)
+            results.extend(result.get(query_schema, []))
+            # Case stop requesting
+            if to_ts_ms == end_ts_ms:
+                break
+            # Case 1 more request: update `to_ts_ms` to `end_ts_ms`
+            elif to_ts_ms + time_delta_ts_ms > end_ts_ms:
+                from_ts_ms = to_ts_ms + 1
+                to_ts_ms = end_ts_ms
+            # Case keep requesting: update `to_ts_ms` to `to_ts_ms` + time delta
+            else:
+                from_ts_ms = to_ts_ms + 1
+                to_ts_ms = to_ts_ms + time_delta_ts_ms
+
+        return results
+
     def query_online_deposits_withdrawals(
             self,
             start_ts: Timestamp,
             end_ts: Timestamp,
     ) -> List[AssetMovement]:
-        # This does not check for any limits. Can there be any limits like with trades
-        # in the deposit/withdrawal binance api? Can't see anything in the docs:
-        # https://github.com/binance-exchange/binance-official-api-docs/blob/master/wapi-api.md#deposit-history-user_data
-        #
-        # Note that all timestamps should be in milliseconds, so we multiply by 1k
-        options = {
-            'timestamp': ts_now_in_ms(),
-            'startTime': start_ts * 1000,
-            'endTime': end_ts * 1000,
-        }
-        result = self.api_query_dict('depositHistory.html', options=options)
-        raw_data = result.get('depositList', [])
-        options['timestamp'] = ts_now_in_ms()
-        result = self.api_query_dict('withdrawHistory.html', options=options)
-        raw_data.extend(result.get('withdrawList', []))
-        log.debug('binance deposit/withdrawal history result', results_num=len(raw_data))
+        """
+        Be aware of:
+          - Timestamps must be in milliseconds.
+          - There must be less than 90 days between start and end timestamps.
+          - Both deposit history and withdraw history do not check for any limits.
```

Comment:
Is this limit comment in reference to my comment from before? I guess it's fine so we can remove it.

Reply:
I found interesting your comment! Unless trades, these endpoints don't have `limit`. And it is just a reminder that if they ever implement limit, the logic would need to be refactored for handling them. I can remove it.